scipy/stats/_continuous_distns.py

```diff
@@ -3377,6 +3377,33 @@ def _cdf(self, x, mu):
     def _stats(self, mu):
         return mu, mu**3.0, 3*np.sqrt(mu), 15*mu
 
+    def fit(self, data, *args, **kwds):
+
+        if type(self) == wald_gen:
+            return super(invgauss_gen, self).fit(data, *args, **kwds)
+
+        data, fshape_s, floc, fscale = _check_fit_input_parameters(self, data,
+                                                                   args, kwds)
+        '''
+        MLE is not used in 3 condtions:
+        - `floc` is not set
+        - `floc` is set but translation results in negative data
+        - `f0` is fixed
+        These three cases fall back on generic optimization.
+        '''
+        if floc is None or fshape_s is not None:
+            return super(invgauss_gen, self).fit(data, *args, **kwds)
+        elif np.any(data - floc < 0):
```

Comment:
In gh-12457 we wrote `np.any(data - floc < (fscale if fscale else 0)`, but here `fscale` is not the minimum of the data. The range of the data is just < 0.